src/blog/understanding-ecmascript-part-3.md

```diff
@@ -0,0 +1,320 @@
+---
+title: 'Understanding the ECMAScript spec, part 3'
+author: '[Marja Hölttä](https://twitter.com/marjakh), speculative specification spectator'
+avatars:
+  - marja-holtta
+date: 2020-02-11 13:33:37
+tags:
+  - ECMAScript
+description: 'Tutorial on reading the ECMAScript specification'
+tweet: ''
+---
+
+... where we dive deep in the syntax!
+
+## Previous episodes
+
+In [part 2](/blog/understanding-ecmascript-part-2), we examined a simple grammar production and how its runtime semantics are defined.  In [the extra content](/blog/extra/understanding-ecmascript-part-2-extra), we also followed a long grammar production chain from `AssignmentExpression` to `MemberExpression`. In this episode, we'll go deeper in the definition of the ECMAScript (or JavaScript) language and its syntax.
+
+If you're not familiar with [context-free grammars](https://en.wikipedia.org/wiki/Context-free_grammar), now it's a good idea to check out the basics, since the spec uses context-free grammars to define the language.
+
+## ECMAScript grammars
+
+ECMAScript source text is a sequence of Unicode code points. Each Unicode code point is an integral value between `U+0000` and `U+10FFFF`. The actual encoding (for example, UTF-8 or UTF-16) is not important &mdash; we assume that the source code has already been converted into a sequence of Unicode code points according to the encoding it was in.
+
+The spec contains several grammars which we'll briefly describe next.
+
+### Lexical grammar
```

Comment:
I think the order of paragraphs is off here in this section. I think it should either be top down or bottom up.

Even though you refer to the Wikipedia article, you can still have a quick introduction here:
As most grammars the lexical grammar consists of a set of production rules which generate tokens for the syntactic grammar. For the lexical grammar, these production rules are called _goal symbols_.

Reply:
Added:

Each grammar is defined as a context-free grammar, consisting of a set of production rules.

----

(The suggestion " For the lexical grammar, these production rules are called goal symbols." is not true; goal symbol is not the a production; each context free grammar has one "main" symbol, such as "Program", which is the goal symbol.)